src/dbnode/namespace/schema_registry.go

```diff
@@ -154,21 +154,13 @@ func (sr *schemaRegistry) RegisterListener(
 	<-watch.C()
 
 	// Deliver the current schema
-	if schema, ok := watchable.Get().(SchemaHistory).GetLatest(); ok {
-		listener.SetSchema(schema)
-	} else if sr.logger != nil {
-		sr.logger.Error("can not update schema listener with empty schema history")
-	}
+	listener.SetSchemaHistory(watchable.Get().(SchemaHistory))
```

Comment:
Curious why you're removing this check from the registry itself. Just to make it more flexible? Otherwise seems prudent to have the check at multiple layers but curious what your thoughts are

Reply:
It is mostly because the listener interface is changed from SetSchema to SetSchemaHistory, so there is no need to call GetLatest here.

Another reason is that the decision that whether or not an empty schema history is valid is better left to the caller to make.

Reply:
ok sure